Automattic/jetpack
# Blocks: Add new Launchpad Modal on Save

#### Changes proposed in this Pull Request:
Currently, there's no easy way for the user to go back to Launchpad after they have made changes in the site editor. This PR creates a Modal that opens every time the user saves a change.
The new modal is currently only in Beta, so only a12s and folks who explicitly turned on Beta extensions will see it.

<img width="1020" alt="image" src="https://user-images.githubusercontent.com/20927667/208516235-4bb523f5-b46e-4532-96b7-871506c0e42f.png">

#### Other information:


#### Jetpack product discussion

* paYE8P-1Kg-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

This can be tested on WordPress.com Simple and WoA sites.

- If on a WordPress.com Simple site, if you are proxied you will have access to this extension right away. On your WordPress.com sandbox, you can run `bin/jetpack-downloader test jetpack update/launchpad-modal-first` to apply the patch and test. More details on this testing process: p9dueE-5Nn-p2
- If on a WoA site, you'll want to add the following to a functionality plugin to turn on Beta extensions: `add_filter( 'jetpack_blocks_variation', function() { return 'beta'; } );`. On any WoA site, you can [install the Beta plugin](https://jetpack.com/download-jetpack-beta/) and switch to the `update/launchpad-modal-first` branch in the plugin settings page to get started with testing.

https://user-images.githubusercontent.com/20927667/208515580-d627b526-8c5a-48b2-9e1b-d854edab22a1.mov

In both scenarios, you'll need a link-in-bio site and a site that hasn't been launched yet.

On a self-hosted site or local instance, you will not see the modal, unless:
1. You activate Beta extensions as explained above.
2. You set the 2 following options (cli commands for our Docker setup):
    - `jetpack docker wp option update launchpad_screen full`
    - `jetpack docker wp option update site_intent link-in-bio`
 
1. Once you are all set with your testing environment, open the site editor (you must be using a block theme), and make a change. 
2. Save your changes and you should see the modal appear.
